Mathlib/Algebra/Module/Torsion.lean

```diff
@@ -873,3 +873,42 @@ theorem isTorsion_iff_isTorsion_int [AddCommGroup M] :
 #align add_monoid.is_torsion_iff_is_torsion_int AddMonoid.isTorsion_iff_isTorsion_int
 
 end AddMonoid
+
+namespace AddSubgroup
+
+variable (A : Type*) [AddCommGroup A] (n : ℕ)
+
+/-- The additive `n`-torsion subgroup for `n` in `ℕ`. -/
+@[reducible]
+def torsionBy := (Submodule.torsionBy ℤ A n).toAddSubgroup
+
+@[inherit_doc]
+scoped notation:max A"["n"]" => torsionBy A n
+
+variable {A n}
+
+lemma mem_torsionBy (x : A) : x ∈ A[n] ↔ n • x = 0 := by simp
+
+@[simp]
+lemma nsmul_torsionBy (x : A[n]) : n • x = 0 :=
+  ZeroMemClass.coe_eq_zero.mp <| (mem_torsionBy _).mp x.property
+
+lemma mod_nsmul_torsionBy_eq_nsmul (m : A[n]) (s : ℕ) :
+    (s % n) • m = s • m := by
+  nth_rewrite 2 [← Nat.div_add_mod s n]
+  rw [add_smul, self_eq_add_left, mul_comm, mul_smul, nsmul_torsionBy, smul_zero]
+
+instance moduleZModTorsionBy [Fact (1 < n)] : Module (ZMod n) A[n] where
```

Comment:
Surely this proof works for 0 < n? And things are also fine for n = 0 but in that case you probably need a different construction.

I am a little unclear about whether introducing `Fact`s is in general a good idea, and whether instead this should just be a def.

Reply:
Yes, I can make it work for n =1, but will require a different argument for `one_smul`. But I've just noticed (`AddCommGroup.zmodModule`)[https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/ZMod/Module.html#AddCommGroup.zmodModule], so I'll just use that instead. Whether this should be a def or instance, I'm not sure.